src/tools/wasm-link.cc

```diff
@@ -50,22 +52,26 @@ static Option s_options[] = {
 };
 WABT_STATIC_ASSERT(NUM_FLAGS == WABT_ARRAY_SIZE(s_options));
 
-typedef const char* String;
-WABT_DEFINE_VECTOR(string, String);
-
 static bool s_debug;
 static bool s_relocatable;
 static const char* s_outfile = "a.wasm";
-static StringVector s_infiles;
+static std::vector<std::string> s_infiles;
 static FileWriter s_log_stream_writer;
 static Stream s_log_stream;
 
 struct Context {
+  WABT_DISALLOW_COPY_AND_ASSIGN(Context);
+  Context();
+
   Stream stream;
-  LinkerInputBinaryVector inputs;
-  ssize_t current_section_payload_offset;
+  std::vector<std::unique_ptr<LinkerInputBinary>> inputs;
+  ssize_t current_section_payload_offset = 0;
 };
 
+Context::Context() {
+  WABT_ZERO_MEMORY(stream);
```

Comment:
I was trying to avoid inlining constructors unless they were trivial (including members that would be default constructed). But yeah, this one looks reasonable to inline.